tests/test_subscriber.py

```diff
@@ -193,19 +195,90 @@ def test_on_put_invalid(self, mock_invalid_param):
         self.assertTrue('invalid_param' in str(context.exception))
 
 
-@patch.object(Auth, 'get_tenants', return_value=None)
-@patch.object(Persister, 'init_mongodb')
 @patch.object(Persister, 'create_indexes_for_notifications')
+@patch.object(Config, 'load_defaults')
 @patch('history.subscriber.persister.Messenger')
-@patch('history.subscriber.persister.falcon.API')
-@patch('history.subscriber.persister.simple_server')
-def test_persister_main(mock_simple_server, mock_falcon_api, mock_messenger, mock_create_indexes_for_notifications,
+def test_persist_all_events(mock_messenger, mock_config, mock_create_index):
+
+    from history.subscriber.persister import start_dojot_messenger
+    from history import conf
+
+    p = Persister()
+    p.create_indexes_for_notifications('admin')
+
+    mock_config.dojot = {
+        "management": {
+            "user": "dojot-management",
+            "tenant": "dojot-management"
+        },
+        "subjects": {
+            "tenancy": "dojot.tenancy",
+            "devices": "dojot.device-manager.device",
+            "device_data": "device-data"
+        }
+    }
+    #test persist all boolean valued events
+
+    conf.dojot_persist_notifications_only = False
+    start_dojot_messenger(mock_config, p)
+
+    #test persist all text valued events
+    conf.dojot_persist_notifications_only = 'False'
+    start_dojot_messenger(mock_config, p)
+
+    assert mock_messenger.called
+    assert mock_create_index.called
+
+
+@patch.object(Persister, 'create_indexes_for_notifications')
+@patch.object(Config, 'load_defaults')
+@patch('history.subscriber.persister.Messenger')
+def test_persist_only_notifications(mock_messenger, mock_config, mock_create_indexes_for_notifications):
+
+    from history.subscriber.persister import start_dojot_messenger
+    from history import conf
+
+    p = Persister()
+    p.create_indexes_for_notifications('admin')
+
+    mock_config.dojot = {
+        "management": {
+            "user": "dojot-management",
+            "tenant": "dojot-management"
+        },
+        "subjects": {
+            "tenancy": "dojot.tenancy",
+            "devices": "dojot.device-manager.device",
+            "device_data": "device-data"
+        }
+    }
+
+    #test persist only boolean valued notifications
+    conf.dojot_persist_notifications_only = True
+    start_dojot_messenger(mock_config, p)
+
+    
+    #test persist only text valued notifications
+    conf.dojot_persist_notifications_only = 'True'
+    start_dojot_messenger(mock_config, p)
+
+    assert mock_messenger.called
+    assert mock_create_indexes_for_notifications.called
```

Comment:
I think this line 266 doesn't make sense since you're explicitly calling it on line 242